contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -1274,6 +1290,16 @@ exec_stmt_exec(PLtsql_execstate *estate, PLtsql_stmt_exec *stmt)
 		if (stmt->is_cross_db)
 			SetCurrentRoleId(current_user_id, false);
 
+		/*
+		 * old_plan_mode may be NULL when an exception occurred before capturing the old mode 
+		 */
+		if ((stmt->exec_with_recompile || created_with_recompile) && !old_plan_mode)
+		{
+			(void) set_config_option("plan_cache_mode", old_plan_mode,
+							  GUC_CONTEXT_CONFIG,
+							  PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);									  		
```

Comment:
If `old_plan_mode` is NULL, that means an exception occurred before setting the config, hence we don't need to restore the config. 

Also, even if `old_plan_mode` is not NULL, we may not need to restore it because we used `GUC_ACTION_SAVE`. But this needs to verified. Do we have a test case for this code path?

Reply:
We are following the same logic here as for setting/resetting the "search_path" GUC.  Also note that when old_plan_mode=NULL, the GUC is not restored.  
I do not see how we can add a test case for this branch. Given that the search_path is GUC is handled this way, we should assume that this code path can be followed and we should be able to properly restore the GUC in that case.

Reply:
This `if` block is executed only when `old_plan_mode` is NULL, and in that case `plan_cache_mode` will be set to its default value (not the previous value, please read the comment about `set_config_option()` function in PG), which is wrong. 

And let me say my point again:
> If old_plan_mode is NULL, that means an exception occurred before setting the config, hence we don't need to restore the config.

I don't see a similar logic regarding setting "search_path" GUC neither. Also, test coverage report shows that 4 lines in this file are not covered, which is not good.